tests/test_optimize.py

```diff
@@ -671,3 +718,149 @@ def test_update_stepsize(self):
         eta2 = 0.1
         opt.update_stepsize(eta2)
         assert opt._stepsize == eta2
+
+
+@pytest.mark.parametrize(
+    "opt, opt_name",
+    [
+        (GradientDescentOptimizer(stepsize), "gd"),
+        (MomentumOptimizer(stepsize, momentum=gamma), "moment"),
+        (NesterovMomentumOptimizer(stepsize, momentum=gamma), "nest"),
+        (AdagradOptimizer(stepsize), "ada"),
+        (RMSPropOptimizer(stepsize, decay=gamma), "rms"),
+        (AdamOptimizer(stepsize, beta1=gamma, beta2=delta), "adam"),
+        (RotosolveOptimizer(), "roto"),
+    ],
+)
+class TestOverOpts:
+    """Tests keywords, multiple arguements, and non-training arguments in relevent optimizers"""
+
+    def test_kwargs(self, opt, opt_name, tol):
+        """Test that the keywords get passed and alter the function"""
```

Comment:
Did you end up getting `mocker.spy` to work, to ensure that the keyword argument gets passed to this function?